### PR TITLE
chore: bump maintenance versions in mergify

### PR DESCRIPTION
**Pull request by mdelapenya**
8.3 is here, let's remove the least minor

## What does this PR do?
It adds 8.3 to mergify, removing 8.1 (prior lowest minor in the same major), and keeping 7.17 as the lowest minor

## Why is it important?
Keep the maintenance branches in sync with the upstream projects

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -18,41 +18,41 @@ pull_request_rules:
...
     actions:
       backport:
         assignees:
           - "{{ author }}"
         branches:
-          - "8.1"
+          - "8.2"
         title: "[{{ destination_branch }}](backport #{{ number }}) {{ title }}"
-  - name: backport patches to 7.17 branch
+  - name: backport patches to 8.1 branch
```
Comment (v1v) [inline .mergify.yml:45]: so `7.17` will not have any default backport, is that the expected behaviour?

Comment (mdelapenya) [reply .mergify.yml:45]: We always move the branch window one level. Although, thinking it twice, we may need to move the 8.minor pointer, not the 7 one. Therefore, we should keep:

7.17 > 8.2 > 8.3 > main

Is this correct?

Comment (v1v) [reply .mergify.yml:45]: I think so, `7.17` is the last `7` minor to be supported until `9` as per https://www.elastic.co/support/eol

Comment (mdelapenya) [reply .mergify.yml]: I kept 7.17, removing 8.1 instead. Thanks for the clarification!

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**